docs/oauth2/grants/grants.rst

```diff
@@ -9,23 +9,30 @@ Grant types
     implicit
     password
     credentials
-    custom_validators
+    refresh
     jwt
+    custom_validators
+    custom_grant
 
-Grant types are what make OAuth 2 so flexible. The Authorization Code grant is
-very similar to OAuth 1 (with less crypto), the Implicit grant serves less
-secure applications such as mobile applications, the Resource Owner Password
-Credentials grant allows for legacy applications to incrementally transition to
-OAuth 2, the Client Credentials grant is excellent for embedded services and
-backend applications.
+Grant types are what make OAuth 2 so flexible. The :doc:`Authorization
+Code grant </oauth2/grants/authcode>` is the default for almost all
+Web Applications, the :doc:`Implicit grant </oauth2/grants/implicit>`
+serves less secure applications such as Mobile Applications or
+Single-Page Applications, the :doc:`Client Credentials grant
+</oauth2/grants/credentials>` is excellent for embedded services and
+backend applications. We have also the :doc:`Resource Owner Password
+Credentials grant </oauth2/grants/password>` for legacy applications
```

Comment:
Why legacy applications?
I use OAuth2 all the time for authentication and authorization of my own users.
This requires username and password.

Reply:
That's the usual recommendation from RFC and BCP. See https://tools.ietf.org/html/rfc6749#section-10.7 

>    The resource owner password credentials grant type is often used for
   legacy or migration reasons.

And BCP at https://tools.ietf.org/html/rfc6819#section-4.4.3 . Also, because the general recommendation of the RFC 

>  The authorization server and client SHOULD minimize use of this grant type and utilize other grant types whenever possible. 

Also, I didn't changed much from the previous iteration. If you'd prefer, we could add and emphasize the following sentence from the RFC:

> The credentials should only be used when there is a high
   degree of trust between the resource owner and the client (e.g., the
   client is part of the device operating system or a highly privileged
   application), and when other authorization grant types are not
   available (such as an authorization code).

Thoughts?

Reply:
Yes, that should explain it better.
"Legacy" implies that one should never use the grant for new applications while in fact you should use the Resource Owner Password
Credentials grant when you are the owner of the application and only then.